networkit/cpp/generators/ConfigurationModel.cpp

```diff
@@ -0,0 +1,93 @@
+#include <networkit/auxiliary/Random.hpp>
+#include <networkit/generators/ConfigurationModel.hpp>
+#include <networkit/generators/HavelHakimiGenerator.hpp>
+#include <networkit/graph/Graph.hpp>
+
+#include <chrono>
+#include <iostream>
+#include <thread>
+
+namespace NetworKit {
+
+ConfigurationModel::ConfigurationModel(const std::vector<count> &sequence)
+    : StaticDegreeSequenceGenerator(sequence) {
+
+    std::sort(seq.begin(), seq.end(), std::greater<count>());
+
+    // Throw an error if the sequence isn't realizable:
+    HavelHakimiGenerator(seq).generate();
```

Comment:
This creates `O(n+m)` temp. memory overhead. Is the Havel Hakimi algorithm faster than calling `isRealizable()` from the base class?